### ISSUE TITLE
Getting CloudFormation error on Auth config updating during amplify push

### ISSUE THREAD

**Issue by mlafo** [aws-amplify/amplify-cli#2893]
**Describe the bug**
After an Amplify push, updating resources in the cloud failed on cognito configuration with the following message :
"Parameters: [userPoolGroupList, resourceNameTruncated, dependsOn] must have values"

**Amplify CLI Version**
4.2.0
Auth is configured with multi-auth : UserPool and Iam

**Comment by mlafo** [comment]
I tried a new installation to get the tree missing parameters.
If added them in the former installation into the auth parameters.json.
After a push, the new error is pretty confusing : 
"Parameters: [resourceNameTruncated, dependsOn, userPoolGroupList] do not exist in the template"

**Comment by mlafo** [comment]
Ok i resolved it.

I tried an Amplify auth update
-> Create or update UserPool
Change nothing to the settings.

But it created a second named "userPoolGroup" witch caused a mess in the next "amplify push" I did.
So i tried a "amplify env pull --restore" that i already did previously but with no success.
And as a miracle the "Auth" line in the next "amplify status" stayed to "update".
And finally a last "amplify push" finished with no error !
